### PR TITLE
Added functions to check etag of elements - integration tests

### PR DESCRIPTION
**Pull request by MorrisJobke**
- downstream https://github.com/owncloud/core/pull/26370

cc @rullzer @nickvergessen

### CODE REVIEW

**Comment by rullzer** [comment]
Mmmm the test fails.

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -245,11 +245,20 @@ public function deleteShare($id) {
 			throw new OCSNotFoundException($this->l->t('could not delete share'));
 		}
 
-		if (!$this->canAccessShare($share, false)) {
+		if (!$this->canAccessShare($share, true)) {
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareAPIController.php]: Well true is the default. Also the parameter was specifically introduced for this case: #637

**Comment by nickvergessen** [comment]
Regarding the last commit, shouldn't you delete the "share exception" instead of the group share, to unshare from self?

- [ ] Smashbox

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 1 above)
Comment (rullzer) [inline apps/files_sharing/lib/Controller/ShareAPIController.php]: Yeah I remember that. It is just that apparently a lot of tests still reply on you to be able to delete received shares via the API.

**Comment by rullzer** [comment]
@nickvergessen ah yes could do that. lets first see if this fixes it then I'll try to make it a bit more beautiful :D

**Comment by rullzer** [comment]
@nickvergessen could you smash? Then this is ready to get in

**Comment by nickvergessen** [comment]
Smashing passed. 👍

**Comment by MorrisJobke** [comment]
:+1:from me as well

**Pull request merged**